From 280ce049ea9f48fb204ffab042f52686607688c6 Mon Sep 17 00:00:00 2001
From: Jan Tulak <jtulak@redhat.com>
Date: Wed, 17 Jan 2018 12:21:03 +0100
Subject: [PATCH 06/22] tests: add a test case for crypt and a weak password

A test case for https://bugzilla.redhat.com/show_bug.cgi?id=1141871 It
depends on the OS configuration, by default, Centos is affected by the
mentioned bug, while Fedora is not.

Signed-off-by: Jan Tulak <jtulak@redhat.com>
---
 tests/bashtests/012-crypt-create.sh | 15 +++++++++++++++
 1 file changed, 15 insertions(+)

diff --git a/tests/bashtests/012-crypt-create.sh b/tests/bashtests/012-crypt-create.sh
index 0b86c1b..2c2befe 100755
--- a/tests/bashtests/012-crypt-create.sh
+++ b/tests/bashtests/012-crypt-create.sh
@@ -103,6 +103,7 @@ ssm remove ${DEV}/$crypt_vol1 ${DEV}/$crypt_vol3 ${DEV}/$crypt_vol2 ${DEV}/$cryp
 # Try non existing extension
 not ssm create -e enigma $dev1
 
+
 # Create encrypted lvm volume
 export SSM_LVM_DEFAULT_POOL=${vg1}_lvm
 export LVOL_PREFIX="lvol"
@@ -167,3 +168,17 @@ check list_table "$(ssm list vol)" $SSM_LVM_DEFAULT_POOL/$lvol1 $SSM_LVM_DEFAULT
 
 ssm remove ${DEV}/$crypt_vol1
 ssm  -f remove $SSM_LVM_DEFAULT_POOL
+
+# Try a short password, one that should not be allowed if password quality is
+# checked. If the OS is configured to allow all passwords, ssm will pass. If
+# not, then ssm should abort before doing anything, and we check if a LV was
+# created (a bug) or not (the correct behaviour).
+
+set +e
+echo -e "a\na" | ssm create $dev1 -e luks
+res=$?
+set -e
+
+if [ $res -ne 0 ]; then
+	not check lv_exists $SSM_LVM_DEFAULT_POOL $lvol1
+fi
-- 
2.21.0

